### PR TITLE
add update and remove which can use limit and order_by

### PR DESCRIPTION
**Pull request by ZerQAQ**
see https://github.com/rbock/sqlpp11/issues/370

### CODE REVIEW

File: include/sqlpp11/mysql/remove.h [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+#include <sqlpp11/using.h>
+#include <sqlpp11/where.h>
+
+namespace sqlpp
+{
+  namespace mysql{
+    struct remove_name_t
+    {
+    };
+    struct remove_t : public statement_name_t<remove_name_t>
```
Comment (rbock) [inline include/sqlpp11/mysql/remove.h]: Can't you just use `sqlpp::remove_t` (same question for `update_t`)?

File: include/sqlpp11/mysql/remove.h [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+      static Context& _(const T& /*unused*/, Context& context)
+      {
+        context << "DELETE";
+
+        return context;
+      }
+    };
+
+    template <typename Database>
+    using blank_remove_t = statement_t<Database, remove_t, no_from_t, no_using_t, no_where_t<true>, no_limit_t, no_order_by_t>;
```
Comment (rbock) [inline include/sqlpp11/mysql/remove.h]: According to mysql's documentation, `ORDER BY` has to come first.

Please add at least one test for remove and another for update.

**Review by rbock — CHANGES_REQUESTED** [review]
Thanks for the pull request.

File: include/sqlpp11/mysql/remove.h [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+#include <sqlpp11/parameter_list.h>
+#include <sqlpp11/prepared_remove.h>
+#include <sqlpp11/noop.h>
+#include <sqlpp11/from.h>
+#include <sqlpp11/using.h>
+#include <sqlpp11/where.h>
+
+namespace sqlpp
+{
+  namespace mysql{
```
Comment (rbock) [inline include/sqlpp11/mysql/remove.h:36]: Is this still needed?

File: include/sqlpp11/mysql/remove.h [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+#include <sqlpp11/statement.h>
+#include <sqlpp11/connection.h>
+#include <sqlpp11/type_traits.h>
+#include <sqlpp11/parameter_list.h>
+#include <sqlpp11/prepared_remove.h>
+#include <sqlpp11/noop.h>
+#include <sqlpp11/from.h>
+#include <sqlpp11/using.h>
+#include <sqlpp11/where.h>
```
Comment (rbock) [inline include/sqlpp11/mysql/remove.h]: The includes need to be cleaned up, too.

Try compiling just the header, to see what is missing (and what is superfluous(.

File: include/sqlpp11/mysql/update.h [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+#include <sqlpp11/single_table.h>
+#include <sqlpp11/update_list.h>
+#include <sqlpp11/noop.h>
+#include <sqlpp11/update.h>
+#include <sqlpp11/where.h>
+
+namespace sqlpp
+{
+  namespace mysql{
+    struct update_name_t
```
Comment (rbock) [inline include/sqlpp11/mysql/update.h:37]: Same in this file.

**Review by rbock — CHANGES_REQUESTED** [review]

File: include/sqlpp11/mysql/remove.h (hunk 3 above)
Comment (rbock) [reply include/sqlpp11/mysql/remove.h:36]: To clarify:
`remove_name_t` and its serializer are not needed, I think. Please remove them (or explain why it is needed).

Similarly for update.

Thanks.

**Comment by zhengminlai** [comment]
Hi, can the pull request be merged?

**Comment by rbock** [comment]
As far as I can tell, `remove_name_t` and `update_name_t` are superfluous and still present. Otherwise, this looks OK to me.

**Comment by rbock** [comment]
Implemented on develop of sqlpp11.

**Pull request closed without merging**